bin/tictactoe

```diff
@@ -1,3 +1,40 @@
-#!/usr/bin/env ruby
+# !/usr/bin/env ruby
 
 require_relative '../config/environment'
+puts ""
+puts "Let's play Tic Tac Toe!"
+puts ""
+input = ""
+while input != "n"
+puts "Pick a number to choose a player mode:"
+puts "| 0 (AI vs AI) | 1 (player vs AI) | 2 (player vs player) |"
+
+input = gets.strip.to_i
+
+case input
+when 0 
+  player_1 = Player::Computer.new("X")
+  player_2 = Player::Computer.new("O")
+when 1
+  puts "Who should go first? (type 'human' or 'AI')"
+  pick = gets.strip
+  if pick == "human"
+    player_1 = Player::Human.new("X")
+    player_2 = Player::Computer.new("O")
+  elsif pick == "AI" || pick == "ai"
+    player_1 = Player::Computer.new("X")
+    player_2 = Player::Human.new("O")
+  end
+when 2
+  player_1 = Player::Human.new("X")
+  player_2 = Player::Human.new("O")
+end
+
+game = Game.new(player_1, player_2)
+game.numbered_board
+game.play
+
+puts "Play again? (type 'y' or 'n')"
+input = gets.strip
+end
+puts "Goodbye!"
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.